### DISCUSSION TITLE
Instructions for adding a new operator to the Community repo

### DISCUSSION THREAD

**Discussion by awnand** [redhat-openshift-ecosystem/community-operators-prod#5376] [Q&A] [answered]
Is there a single set of pages with all instructions for adding a new Operator? The [README](https://github.com/redhat-openshift-ecosystem/community-operators-prod/blob/main/README.md) feels incomplete. It links to [this page](https://redhat-openshift-ecosystem.github.io/operator-pipelines/users/contributing-prerequisites/) with some instructions, but the [where-to-contribute](https://redhat-openshift-ecosystem.github.io/operator-pipelines/users/contributing-where-to/) page for example does not seem to have instructions on populating the `ci.yaml` or the OpenShift YAML files. 

For more context, I am trying to add an Operator that I created not using operator-sdk.

**Comment by Allda** [comment, marked as answer]
@awnand Hey, we recently updated the doc with a guide on contributing to the repository. The [where-to-contribute](https://redhat-openshift-ecosystem.github.io/operator-pipelines/users/contributing-where-to/) is the right place to go. There is also a link to a `ci.yaml` description with all supported options.

**Reply by awnand** [reply]
Thanks @Allda for the answer!

For those that are in a similar boat as well, I thought I'd add context for what I meant by "trying to add an Operator that I created not using operator-sdk". The operator was developed using kubebuilder. As a deployment mechanism we offer a Helm chart that deploys the operator Deployment and installs CRDs. We used the following to create the base of our ClusterServiceVersion:

- `helm template` to render the manifests from the Helm chart with the values.
- [`operator-sdk generate bundle`](https://sdk.operatorframework.io/docs/cli/operator-sdk_generate_bundle/) specifically with the `--crds-dir`  `--deploy-dir` options with the files from the rendered Helm chart.

After that, we have a Bundle directory with a ClusterServiceVersion that is partially automatically filled out. We then followed [OpenShift docs](https://docs.openshift.com/container-platform/4.17/operators/operator_sdk/osdk-generating-csvs.html#osdk-manually-defined-csv-fields_osdk-generating-csvs) and other soruces to fill out the other fields.
